modal/experimental/__init__.py

```diff
@@ -209,3 +213,64 @@ async def update_autoscaler(
 
     request = api_pb2.FunctionUpdateSchedulingParamsRequest(function_id=f.object_id, settings=settings)
     await retry_transient_errors(client.stub.FunctionUpdateSchedulingParams, request)
+
+
+class _FlashManager:
+    def __init__(self, client: _Client, port: int):
+        self.client = client
+        self.port = port
+        self.tunnel_manager = _forward_tunnel(port, client=client)
+
+    async def _start(self):
+        tunnel = await self.tunnel_manager.__aenter__()
+
+        host, port = tunnel.url.split("://")[1].split(":")
+        self.heartbeat_task = asyncio.create_task(self._run_heartbeat(host, int(port)))
+
+    async def _run_heartbeat(self, host: str, port: int):
+        first_registration = True
+        while True:
+            try:
+                resp = await self.client.stub.FlashContainerRegister(
+                    api_pb2.FlashContainerRegisterRequest(
+                        priority=10,
+                        weight=5,
+                        host=host,
+                        port=port,
+                    ),
+                    timeout=1,
+                )
+                if first_registration:
+                    logger.warning(f"[Modal Flash] Listening at {resp.url}")
+                    first_registration = False
+            except asyncio.CancelledError:
+                logger.warning("[Modal Flash] Shutting down...")
+                break
+            except Exception as e:
+                logger.error(f"[Modal Flash] Heartbeat failed: {e}")
+
+            try:
+                await asyncio.sleep(1)
+            except asyncio.CancelledError:
+                logger.warning("[Modal Flash] Shutting down...")
+                break
+
+    async def stop(self):
+        self.heartbeat_task.cancel()
+        await retry_transient_errors(
+            self.client.stub.FlashContainerDeregister,
+            api_pb2.FlashContainerDeregisterRequest(),
+        )
+        await self.tunnel_manager.__aexit__(*sys.exc_info())
+
+
+FlashManager = synchronize_api(_FlashManager)
+
+
+@synchronizer.create_blocking
+async def flash_forward(port: int) -> _FlashManager:
```

Comment:
Something I've been trying to do with all the `modal.experimental` functions I've added is to set clear expectations in the docstring about what's experimental. Is this functionality we know we're going to add, but we're not sure about the client API? Or is it experimental _functionality_ and could be dropped entirely, etc. What is the timeline for graduating out of experimental (if known), etc. Basically we want users to understand if this is something they should rely on or not.